ui/src/main/java/edu/wpi/grip/ui/DeployController.java

```diff
@@ -0,0 +1,189 @@
+package edu.wpi.grip.ui;
+
+import com.google.common.eventbus.EventBus;
+import com.google.common.eventbus.Subscribe;
+import com.google.common.io.LineReader;
+import edu.wpi.grip.core.Pipeline;
+import edu.wpi.grip.core.events.ProjectSettingsChangedEvent;
+import edu.wpi.grip.core.events.StopPipelineEvent;
+import edu.wpi.grip.core.serialization.Project;
+import edu.wpi.grip.ui.annotations.ParametrizedController;
+import edu.wpi.grip.ui.util.StringInMemoryFile;
+import javafx.application.Platform;
+import javafx.beans.property.BooleanProperty;
+import javafx.beans.property.SimpleBooleanProperty;
+import javafx.fxml.FXML;
+import javafx.scene.control.*;
+import net.schmizz.sshj.SSHClient;
+import net.schmizz.sshj.common.IOUtils;
+import net.schmizz.sshj.common.StreamCopier;
+import net.schmizz.sshj.connection.channel.direct.Session;
+import net.schmizz.sshj.userauth.UserAuthException;
+import net.schmizz.sshj.xfer.FileSystemFile;
+import net.schmizz.sshj.xfer.LoggingTransferListener;
+import net.schmizz.sshj.xfer.scp.SCPFileTransfer;
+
+import javax.inject.Inject;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.StringWriter;
+import java.net.UnknownHostException;
+import java.util.Optional;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ * JavaFX controller for the deploy tool.
+ * <p>
+ * This basically uploads a headless version of GRIP and the current project to a remote target using SSH, then
+ * runs it remotely.  The default values for all fields are based on typical settings for FRC, with the address
+ * based on the project settings.
+ */
+@ParametrizedController(url = "Deploy.fxml")
+public class DeployController {
+
+    // Default deploy information for FRC. This can be overridden by the use for applications outside of FRC or not
+    // using the roboRIO (ie: coprocessors)
+    public final static String DEFAULT_USER = "lvuser";
+    private final static String DEFAULT_PASSWORD = "";
+    private final static String DEFAULT_JAVA_HOME = "/usr/local/frc/JRE/";
+    private final static String DEFAULT_DIR = "/home/" + DEFAULT_USER;
+    private final static String GRIP_JAR = "grip.jar";
+    private final static String PROJECT_FILE = "project.grip";
+
+    @FXML private TextField address;
+    @FXML private TextField user;
+    @FXML private TextField password;
```

Comment:
Use a password field for this.
https://docs.oracle.com/javase/8/javafx/api/javafx/scene/control/PasswordField.html